### PR TITLE
Add a way to specify description when creating github-like siblings

### LINKED ISSUE #6816: Is there a way to manually change the repo description via create-sibling-github?
https://github.com/datalad/datalad/blob/9cec945af3acebd319825ef9570e74e187596ace/datalad/distributed/create_sibling_ghlike.py#L454

Currently it's set to 'some default' by default. I plan to use gh to edit this after pushing to github, but it would be nice to be able to do this with a flag in create-sibling-github.

### PR DESCRIPTION
**Pull request by mslw**
This PR adds an optional `description` parameter to `create_sibling_ghlike`, and enables it for related (gin / gitea / github / gogs) `create-sibling` commands. This optional parameter allows specification of repository description (displayed on the project's page), replacing the hardcoded 'some default' (which would still be used the default).

Manually tested for Gin and GitHub.

Fixes #6816

### CODE REVIEW

File: changelog.d/20221024_144940_m.szczepanik_nf_ghlike_description.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+### Enhancements and New Features
+
+- A repository description can be specified with a new `--description`
+  option when creating siblings using `create-sibling-[gin|gitea|github|gogs]`.
+  Fixes [#6816](https://github.com/datalad/datalad/issues/6816)
+  via [PR #YYY](https://github.com/datalad/datalad/pull/7109)
```
Comment (yarikoptic) [inline changelog.d/20221024_144940_m.szczepanik_nf_ghlike_description.md:6]:
```suggestion
  via [PR #7109](https://github.com/datalad/datalad/pull/7109)
```

Comment (yarikoptic) [reply changelog.d/20221024_144940_m.szczepanik_nf_ghlike_description.md:6]: the curse of doing things manually which should be automated ;)

**Comment by yarikoptic** [comment]
travis errors are due to odd behavior of `pip` installing older version of datalad-installer

```
0.81s$ pip install datalad-installer
Collecting datalad-installer
  Downloading datalad_installer-0.5.4-py3-none-any.whl (22 kB)
Installing collected packages: datalad-installer
Successfully installed datalad-installer-0.5.4
0.07s$ eval datalad-installer --sudo ok -E new.env ${_DL_ANNEX_INSTALL_SCENARIO}
Usage: datalad-installer [<options>] miniconda [<options>]
option --python-match not recognized
```

(current only is v0.9.1-7-ge3aa0ee) for pythons 3.8, 3.9, 3.10

 @jwodder -- any idea on why it could have happened all of a sudden?

**Comment by jwodder** [comment]
@yarikoptic There was an issue in the past hour with PyPI backends being unavailable; see https://status.python.org/#past-incidents.  Can you rerun the builds?

**Comment by yarikoptic** [comment]
Thank you @jwodder !  Finally travis let me to login (can't before earlier today) so I restarted travis build.

**Comment by bpoldrack** [comment]
Restarted extension test that should be fixed by now.
Ignore the failure on macOS - this is unrelated. Trial to rectify this is #7126

**Comment by bpoldrack** [comment]
Ok, passed.

Thx, @mslw

**Comment by yarikoptic-gitmate** [comment]
PR released in [`0.18.0`](https://github.com/datalad/datalad/releases/tag/0.18.0)

**Pull request merged**